### PR TITLE
Handle relative redirects.

### LINKED ISSUE #2259: HttpClient does not handle redirects properly
I'm using (actually trying to use) a maven repository that answers some requests with relative redirects ("`Location: <relative URL>`"). This complies to the [latest standard](https://tools.ietf.org/html/rfc7231#section-7.1.2). The HttpClient in bnd currently only handles absolute URLs in this header.
 
The error is:

`java.net.MalformedURLException: no protocol: /com/...`

which is caused by calling `new URL(location)` with `location` being (only) a path (i.e. a relative URL).

### PR DESCRIPTION
**Pull request by mnlipp**
Handle relative redirects. Closes #2259.

Signed-off-by: Michael Lipp <mnl@mnl.de>

### CODE REVIEW

File: biz.aQute.http.testservers/src/aQute/http/testservers/Httpbin.java [hunk 1]
```diff
@@ -166,7 +166,9 @@ public void _index(Request rq, Response rsp) throws IOException {
 	public void _redirect(Request rq, Response rsp, int count, int response) throws URISyntaxException {
 		if (count > 0) {
 			System.out.println("redirect " + count);
-			rsp.headers.put("Location", getBaseURI() + "/redirect/" + (--count) + "/" + response);
+			String location = (!rq.args.containsKey("relative") ? getBaseURI() : "") + "/redirect/" + (--count) + "/"
```
Comment (bjhargrave) [inline biz.aQute.http.testservers/src/aQute/http/testservers/Httpbin.java]: Shouldn't this be more than just `containsKey` and instead check the value for truthiness? In the test case you use `?relative=true`. So there seems a small disconnect there.

Comment (mnlipp) [reply biz.aQute.http.testservers/src/aQute/http/testservers/Httpbin.java]: There are no other possible values for key "relative". But AFAIK, the syntax for the query part must be "key=value", even if the value doesn't matter.

**Comment by bjhargrave** [comment]
And now the AppVeyor build works!

File: biz.aQute.http.testservers/src/aQute/http/testservers/Httpbin.java (hunk 1 above)
Comment (bjhargrave) [reply biz.aQute.http.testservers/src/aQute/http/testservers/Httpbin.java]: An alternate would be a different path `/redirectrelative/`

**Comment by bjhargrave** [comment]
Thanks for the PR!

File: biz.aQute.http.testservers/src/aQute/http/testservers/Httpbin.java (hunk 1 above)
Comment (mnlipp) [reply biz.aQute.http.testservers/src/aQute/http/testservers/Httpbin.java]: Maybe, I'm not as deep in your test server design as you. But I don't like duplicating code and this seemed the easiest way to modify the behavior of _redirect in that particular aspect.

**Pull request merged**